src/grt/src/GlobalRouter.cpp

```diff
@@ -2227,6 +2227,131 @@ void GlobalRouter::readSegments(const char* file_name)
           GRT, 261, "Error reading global route segments file {}.", file_name);
     }
   }
+  for (auto& [net, segments] : routes_) {
+    if (!isConnected(net)) {
+      logger_->error(
+          GRT, 262, "Net {} has disconnected segments.", net->getName());
+    }
+    for (const Pin& pin : db_net_map_[net]->getPins()) {
+      bool pin_is_covered = false;
+      for (const GSegment& seg : segments) {
+        // Pin is horizontally covered by the segment
+        if (pin.getOnGridPosition().getX() >= std::min(seg.init_x, seg.final_x)
+            && pin.getOnGridPosition().getX()
+                   <= std::max(seg.init_x, seg.final_x)
+            // Pin is vertically covered by the segment
+            && pin.getOnGridPosition().getY()
+                   >= std::min(seg.init_y, seg.final_y)
+            && pin.getOnGridPosition().getY()
+                   <= std::max(seg.init_y, seg.final_y)
+            // Pin and segment share a layer
+            && pin.getConnectionLayer()
+                   >= std::min(seg.init_layer, seg.final_layer)
+            && pin.getConnectionLayer()
+                   <= std::max(seg.init_layer, seg.final_layer)) {
+          pin_is_covered = true;
+          break;
+        }
+      }
+      if (!pin_is_covered) {
+        logger_->error(GRT,
+                       263,
+                       "Pin {} is not covered by net {}",
+                       pin.getName(),
+                       net->getName());
+      }
+    }
+  }
```

Comment:
Turn this code into a separate _netIsCovered_ function. The function can receive the grt::Net* and the GRoute& as parameters.